### PR TITLE
#2 - eloquent helper

### LINKED ISSUE #2: Eloquent helper
I would like to see a way to provide Eloquent objects into wrapper:

```php
$builder = new LaravelHeatmapBuilder();
$result = $builder->build(Book::class, "read_at");
```

Or:
```php
$builder = new LaravelHeatmapBuilder();
$result = $builder->build(Book::query()->whereYear("read_at", Carbon::now()->year), "read_at");
```

Or:
```php
$builder = new LaravelHeatmapBuilder();
$result = $builder->build(Book::query()->whereYear("read_at", Carbon::now()->year)->get(), "read_at");
```

Or all three possible.

### PR DESCRIPTION
**Pull request by BarTracz**
Builder now allows to build heatmaps using eloquent models and collections. This should close #2

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        foreach ($heatmap as $tile) {
+            echo "<div class='w-12 h-12 {$tile->description} text-black rounded flex items-center justify-center border border-gray-300'></div>";
+        }
+
+        file_put_contents('heatmapOutput.json', json_encode($heatmap));
+        ?>
+    </div>
+</div>
+</body>
+</html>
```
Comment (kamilpiech97) [inline index.php]:
```suggestion
</html>

```

File: src/LaravelHeatmapBuilder.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        );
+    }
+
+    protected function mapToCarbon(array|ArrayAccess|TimeGroupable|Model $item): Carbon
+    {
+        if ($item instanceof TimeGroupable) {
+            $date = Carbon::parse($item->getTimeGroupableIndicator());
+        } elseif ($item instanceof Model) {
+            if (!isset($item->{$this->arrayAccessIndex})) {
+                throw new \InvalidArgumentException("The property '{$this->arrayAccessIndex}' does not exist on the given model.");
```
Comment (kamilpiech97) [inline src/LaravelHeatmapBuilder.php]:
```suggestion
                throw new \InvalidArgumentException("The property '{$this->arrayAccessIndex}' does not exist on the given model.");
```
lets import that

File: index.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        $jsonData = file_get_contents('data.json');
+        $data = json_decode($jsonData, true);
+
+        $now = Carbon::now();
+        $interval = PeriodInterval::Daily;
+
+        $startOfMonth = $now->copy()->startOfMonth();
+        $endOfMonth = $now->copy()->endOfMonth();
+
+        $now = Carbon::now();
```
Comment (kamilpiech97) [inline index.php]:
```suggestion
```

File: index.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Heatmap Display</title>
+    <link href="https://cdn.jsdelivr.net/npm/tailwindcss@2.2.19/dist/tailwind.min.css" rel="stylesheet">
+</head>
```
Comment (kamilpiech97) [inline index.php]: For me better should be write tests for that.

Comment (kamilpiech97) [reply index.php]: That `index.php`, you can use locally if you need

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

**Comment by BarTracz** [comment]
This index.php shouldn`t be there, there is one in php version already

**Comment by BarTracz** [comment]
Now instead of 1 builder with many ifs to determine which operation to make, method is separated to buildFromArray, and buildFromCollection to make it easier to add new accepted data types.

**Comment by BarTracz** [comment]
Tests can be added in #4(which i plan to do just after this issue), so i guess it is fine to close it without implementing any.

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**